crates/typed-sled/.gitignore

```diff
@@ -0,0 +1 @@
+/target
```

Comment:
Why do we need this? Can't we update the https://github.com/alpenlabs/alpen/blob/fcae36d817f7195ae34ad1e616c507f793e46cdd/.gitignore#L62

to be:

```
**/target/
``

Reply:
Yes, but since this is supposed to go in the separate repo, I wanted to make this crate as independent to workspace as possible.